### PR TITLE
YLP-2354 Access device settings from tabs

### PR DESCRIPTION
**Pull request by fatb38**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

**Comment by TimCanu** [comment]
Screenshot from 2023-06-15 08-34-36
Is it normal that we still have this?

File: packages/yourloops/test/integration/assert/device-page.assert.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  Object.assign(navigator, {
+    clipboard: {
+      writeText
+    }
+  })
+  const deviceSettings = screen.getByTestId('device-settings-container')
+  const copyTextButton = within(deviceSettings).getByRole('button', { name: 'Copy as text' })
+  await userEvent.click(copyTextButton)
+
+  expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1)
```
Comment (TimCanu) [inline packages/yourloops/test/integration/assert/device-page.assert.ts:47]:
```suggestion
  const copiedStringToPaste = '01/01/2020, 00:00:00\n\n-- DBL --\nManufacturer      Diabeloop\nIdentifier        1234\nIMEI              1234567890\nSoftware version  1.0.5.25\n\n-- Settings --\nName    Value  Unit\nWEIGHT   72.0  kg'
  expect(navigator.clipboard.writeText).toHaveBeenCalledWith(copiedStringToPaste)
```

:warning: The date I gave probably does not work on your machine.
A lesser solution (to having the date working on every machine) could:
```
const copiedStringToPaste = '/2020, 00:00:00\n\n-- DBL --\nManufacturer      Diabeloop\nIdentifier        1234\nIMEI              1234567890\nSoftware version  1.0.5.25\n\n-- Settings --\nName    Value  Unit\nWEIGHT   72.0  kg'
expect(navigator.clipboard.writeText).toHaveBeenCalledWith(expect.stringContaining(copiedStringToPaste))
```

File: packages/yourloops/test/integration/mock/data.api.mock.ts [hunk 2]
```diff
@@ -194,6 +194,28 @@ export const twoWeeksOfCbg: Data = {
...
+    { time: '2020-01-09T10:00:00Z', type: 'cbg', id: '2020-01-09_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' },
+    { time: '2020-01-10T10:00:00Z', type: 'cbg', id: '2020-01-10_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' },
+    { time: '2020-01-11T10:00:00Z', type: 'cbg', id: '2020-01-11_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' },
+    { time: '2020-01-12T10:00:00Z', type: 'cbg', id: '2020-01-12_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' },
+    { time: '2020-01-13T10:00:00Z', type: 'cbg', id: '2020-01-13_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' },
+    { time: '2020-01-14T10:00:00Z', type: 'cbg', id: '2020-01-14_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' },
+    { time: '2020-01-15T10:00:00Z', type: 'cbg', id: '2020-01-15_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' }
```
Comment (TimCanu) [inline packages/yourloops/test/integration/mock/data.api.mock.ts:215]:
```suggestion
  dataRange: ['2020-01-01T16:35:00Z', '2020-01-01T16:40:00Z'],
  data: [
    { time: '2020-01-01T10:00:00Z', type: 'pumpSettings', id: 'pump_settings', timezone: 'UTC', uploadId: 'osef', payload: { basalsecurityprofile: null, cgm: { apiVersion: 'v1', endOfLifeTransmitterDate: '2050-04-12T17:53:54+02:00', expirationDate: '2050-04-12T17:53:54+02:00', manufacturer: 'Dexcom', name: 'G6', swVersionTransmitter: 'v1', transmitterId: 'a1234' }, device: { deviceId: '1234', imei: '1234567890', manufacturer: 'Diabeloop', name: 'DBLG1', swVersion: '1.0.5.25' }, history, parameters: [{ name: 'WEIGHT', value: '72', unit: 'kg', level: 1, effectiveDate: '2020-01-01T10:00:00Z' }], pump: { expirationDate: '2050-04-12T17:53:54+02:00', manufacturer: 'Vicentra', name: 'Kaleido', serialNumber: '123456', swVersion: 'beta' } }, _userId: 'osef' },
    { time: '2020-01-01T10:00:00Z', type: 'cbg', id: '2020-01-01_0', timezone: 'Europe/Paris', units: 'mmol/L', value: 10.5, uploadId: 'osef', _userId: 'osef' }
```

File: packages/yourloops/test/integration/pages/device/device-page.caregiver.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    mockPatientApiForCaregivers()
+    mockDataAPI(pumpSettingsData)
+  })
+
+  it('should render correct components when navigating to the device page', async () => {
+    let router
+    await act(async () => {
+      router = renderPage(`/patient/${patient1Id}/device`)
+    })
+    checkDeviceSettingsContent()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/device/device-page.caregiver.spec.tsx:60]: This does not follow the new method naming/architecture that we decided for ITs.
You should call a method name `testDeviceSettingVisualisation` (something like that) defined in a file `use-cases/device-settings-visualisation.ts` that call the method `checkDeviceSettingsContent`.
You could also call `testAppMainLayoutForCaregiver` in here.

Similar comment for the method `checkNavigationToDailyView`.

Alternatively, you can have one `it` per `testSomething` called.

```
it('should render correct layout', async () => {
    let router
    await act(async () => {
      router = renderPage(`/patient/${patient1Id}/device`)
    })
    await testAppMainLayoutForCaregiver()
  })

  it('should display correct parameters', async () => {
    let router
    await act(async () => {
      router = renderPage(`/patient/${patient1Id}/device`)
    })
    testDeviceSettingsContent()
    await checkNavigationToDailyView(router, `/patient/${patient1Id}/daily`)
  })

  it('should navigate to daily page when clicking on the daily button', async () => {
    let router
    await act(async () => {
      router = renderPage(`/patient/${patient1Id}/device`)
    })
    await testNavigationToDailyView(router, `/patient/${patient1Id}/daily`)
  })
```

File: packages/yourloops/test/integration/pages/device/device-page.hcp.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    mockAuth0Hook()
+    mockNotificationAPI()
+    mockDirectShareApi()
+    mockTeamAPI()
+    mockUserApi().mockUserDataFetch({ firstName, lastName })
+    mockPatientApiForHcp()
+    mockDataAPI(pumpSettingsData)
+  })
+
+  it('should render correct components when navigating to the device page', async () => {
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/device/device-page.hcp.spec.tsx:60]: Same comment

File: packages/yourloops/test/integration/pages/device/device-page.patient.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    mockAuth0Hook(UserRole.Patient, patient1Id)
+    mockNotificationAPI()
+    mockDirectShareApi()
+    mockTeamAPI()
+    mockUserApi().mockUserDataFetch({ firstName, lastName })
+    mockPatientApiForPatients()
+    mockDataAPI(pumpSettingsData)
+  })
+
+  it('should render correct components when navigating to the device page', async () => {
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/device/device-page.patient.spec.tsx:57]: Same comment

File: packages/yourloops/components/device/device-settings.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  goToDailySpecificDate: (date: number | Date) => void
+  pumpSettings: PumpSettings
+  timePrefs: TimePrefs
+}
+
+export const DeviceSettings: FC<DeviceSettingsProps> = ({ pumpSettings, timePrefs, goToDailySpecificDate }) => {
+  const { t } = useTranslation()
+  const { device, pump, cgm, parameters, history } = pumpSettings.payload
+
+  // TODO Set this one directly in the futur new component (see YLP-2447 https://diabeloop.atlassian.net/browse/YLP-2354)
```
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/components/device/device-settings.tsx:49]: With the following feature (that is to change the page design), I plan to write a custom hook to handle states and futur new components props. My message is not very clear I agree, but this one will be wrapped into a function in a custom hook ;)

File: packages/yourloops/components/device/device-settings.utils.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { type ParametersChange, Unit } from 'medical-domain'
+
+export const formatParameterValue = (value: string | number, units: string | Unit): string => {
```
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/components/device/device-settings.utils.ts:30]: Yes old code, I tried to clean it a little bit already. But some lines are so mysterious...

**Comment by fatb38** [comment]
> Screenshot from 2023-06-15 08-34-36 Is it normal that we still have this?

_Evanesco_

**Review by TimCanu — APPROVED** [review]

**Pull request merged**